### PR TITLE
N2D (AMD Rome) support in GCP

### PR DESCRIPTION
**Pull request by KevinDuringWork**
Please consider supporting custom N2D instances (AMD Rome) under the cpuPlatform option for the Google Cloud.

### CODE REVIEW

File: centaur/src/main/resources/standardTestCases/papi_cpu_platform/papi_cpu_platform.wdl [hunk 1]
```diff
@@ -27,4 +27,5 @@ workflow cpus {
    call cpu_platform as haswell { input: cpu_platform = "Intel Haswell" }
    call cpu_platform as broadwell { input: cpu_platform = "Intel Broadwell" }
    call cpu_platform as cascadeLake { input: cpu_platform = "Intel Cascade Lake" }
+   call cpu_platform as rome {input: cpu_platform = "AMD Rome" }
```
Comment (mcovarr) [inline centaur/src/main/resources/standardTestCases/papi_cpu_platform/papi_cpu_platform.wdl:30]: there should be a corresponding assertion in `papi_cpu_platform.test`

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/CustomMachineType.scala [hunk 2]
```diff
@@ -159,3 +160,24 @@ case object N2CustomMachineType extends CustomMachineType {
...
+  override val minMemoryPerCpu: MemorySize = MemorySize(0.5, MemoryUnit.GB)
+  override val maxMemoryPerCpu: MemorySize = MemorySize(8.0, MemoryUnit.GB)
+  override val memoryFactor: MemorySize = MemorySize(256, MemoryUnit.MB)
+  
+  override def validateCpu(cpu: Refined[Int, Positive]): Int = {
+    cpu.value match {
+      case cpu if cpu <= 16 => 2 max pow(2, (log(cpu.toDouble)/log(2)).ceil).toInt
+      case cpu if cpu > 16 && cpu <= 96 && cpu % 16 == 0 => cpu
+      case cpu if cpu > 16 && cpu <= 96 => cpu + 16 - (cpu % 16)
+      case cpu if cpu > 96 => 96
```
Comment (mcovarr) [inline supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/CustomMachineType.scala:175]: codecov appears to be correct on this point...

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/MachineConstraints.scala [hunk 3]
```diff
@@ -20,6 +20,7 @@ object MachineConstraints {
       val customMachineType =
```
Comment (mcovarr) [inline supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/MachineConstraints.scala:20]: It reads a little strange that the comment on line 19 only mentions Cascade Lake when something similar is now also happening with AMD Rome

**Review by mcovarr — COMMENTED** [review]
I would also add a CHANGELOG.md entry like what was added for Cascade Lake support in Cromwell 64

**Review by aednichols — APPROVED** [review]

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/CustomMachineType.scala (hunk 2 above)
Comment (KevinDuringWork) [reply supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/CustomMachineType.scala:175]: Will add to tests.

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/MachineConstraints.scala (hunk 3 above)
Comment (KevinDuringWork) [reply supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/MachineConstraints.scala:21]: Updated comment with regards to AMD Rome

Comment (KevinDuringWork) [reply supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/MachineConstraints.scala:21]: test cases added for CPU specified 81 => 96 and 128 => 96.

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/CustomMachineType.scala (hunk 2 above)
Comment (KevinDuringWork) [reply supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/CustomMachineType.scala:175]: Test cases added for CPU specified 81 => 96 and 128 => 96.

**Review by mcovarr — APPROVED** [review]
LGTM assuming all tests pass. I kicked off a full CI run on a clone branch here: https://travis-ci.com/github/broadinstitute/cromwell/builds/231552209

**Comment by mcovarr** [comment]
Thank you for the contribution!

**Pull request merged**